sotodlib/io/bookbinder.py

```diff
@@ -1125,3 +1196,105 @@ def __call__(self):
+        """
+        files = []
+        for subdir in os.listdir(self.root_dir):
+            try:
+                tcode = int(subdir)
+            except:
+                continue
+
+            if not (((start//1e5) - 1) <= tcode <= stop//1e5):
+                continue
+
+            subpath = os.path.join(self.root_dir, subdir)
+            files.extend([os.path.join(subpath, f) for f in os.listdir(subpath)])
+
+        files = np.array(sorted(files))
+        file_times = np.array([int(os.path.basename(f).split('.')[0]) for f in files])
+
+        m = (start <= file_times) & (file_times < stop)
+        if not np.any(m):
+            self.files = []
+            return self.files
+
+        # Add file before for good measure
+        i0 = np.where(m)[0][0]
+        if i0 > 0:
+            m[i0 - 1] = 1
+
+        self.files = files[m].tolist()
+        return self.files
+
+    def frame_iter(self):
+        """Iterator that loops through G3 Frames of all files in self.files"""
+        return itertools.chain(*[core.G3File(f) for f in self.files])
+
+    def load_data(self, start, stop):
+        self.get_files(start, stop)
+
+        data = {}  # structured data[block_name][field]
+        times = {}  # structured times[block_name]
+
+        for frame in self.frame_iter():
+            if frame['hkagg_type'] != so3g.HKFrameType.data:
+                continue
+
+            for bname, block in zip(frame['block_names'], frame['blocks']):
+                if bname not in data:
+                    data[bname] = {
+                        k: [] for k in block.keys()
+                    }
+                    times[bname] = []
+
+                ts = np.array(block.times) / core.G3Units.s
+
+                # Includes a sec before and after for interpolation
+                m = ((start - 1) < ts) & (ts < (stop + 1))
+
+                for field, d in block.items():
+                    data[bname][field].append(np.array(d)[m])
+                times[bname].append(ts[m])
+
+        for bname in data:
+            for field in data[bname]:
+                data[bname][field] = np.hstack(data[bname][field])
+            times[bname] = np.hstack(times[bname])
+
+        self.data = data
+        self.times = times
+        if len(data) != 0:
+            self.hwp_angle_interp = interp1d(
+                times['fast'], np.unwrap(data['fast']['hwp_angle']), 
+                bounds_error=False)
+            self.locked_interp = interp1d(
+                times['slow'], data['slow']['locked'], bounds_error=False)
+            self.stable_interp = interp1d(
+                times['slow'], data['slow']['stable'], bounds_error=False)
+            self.hwp_rate_interp = interp1d(
+                times['slow'], data['slow']['hwp_rate'], bounds_error=False)
+        else:
+            self.hwp_angle_interp = None
+            self.locked_interp = None
+            self.stable_interp = None
+            self.hwp_rate_interp = None
```

Comment:
Do we anticipate needing this elsewhere? If so, might be good as a separate module/file in the sotodlib/io/ directory.

I usually put a newline at the end of my files out of habit, but I guess that’s not strictly necessary.

Reply:
Ah I'm too spoiled by the autoformatter that Brian added in ocs/socs haha. I'll add it. I'm not sure how useful it is elsewhere in sotodlib, so I kind of think we should just keep it here for now since we can easily move it to a separate module later if that ends up being helpful